src/Form/MediaLibraryEmbeddableForm.php

```diff
@@ -0,0 +1,171 @@
+    $ajax_query += [FormBuilderInterface::AJAX_FORM_REQUEST => TRUE];
+    $form['container']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Add'),
+      '#button_type' => 'primary',
+      '#validate' => ['::validateEmbeddable'],
+      '#submit' => ['::addButtonSubmit'],
+      '#ajax' => [
+        'callback' => '::updateFormCallback',
+        'wrapper' => 'media-library-wrapper',
+        // Add a fixed URL to post the form since AJAX forms are automatically
+        // posted to <current> instead of $form['#action'].
+        // @todo Remove when https://www.drupal.org/project/drupal/issues/2504115
+        //   is fixed.
+        'url' => Url::fromRoute('media_library.ui'),
+        'options' => ['query' => $ajax_query],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Informs us if we are working with an unstructured embed.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current form state.
+   * @return bool
+   *   True if unstructured, otherwise false.
+   */
+  public function isUnstructured(FormStateInterface $form_state) {
+    $embed_code = $form_state->getValue('field_media_embeddable_code');
+    if ($embed_code) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Validates the oEmbed URL.
+   *
+   * @param array $form
+   *   The complete form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current form state.
+   */
+  public function validateEmbeddable(array &$form, FormStateInterface $form_state) {
+    // No validation necessary if we have an embed code.
+    if ($this->isUnstructured($form_state)) {
+      return;
+    }
+    parent::validateUrl($form, $form_state);
+  }
+
+  /**
+   * Submit handler for the add button.
+   *
+   * @param array $form
+   *   The form render array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function addButtonSubmit(array $form, FormStateInterface $form_state) {
+    if ($this->isUnstructured($form_state)) {
+      $values = [$form_state->getValue('field_media_embeddable_code')];
+    } else {
+      $values = [$form_state->getValue('field_media_embeddable_oembed')];
+    }
+    $this->processInputValues($values, $form, $form_state);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function processInputValues(array $source_field_values, array $form, FormStateInterface $form_state) {
+    $media_type = $this->getMediaType($form_state);
+    $media_storage = $this->entityTypeManager->getStorage('media');
+    if ($this->isUnstructured($form_state)) {
+      $source_field_name = 'field_media_embeddable_code';
+    } else {
+      $source_field_name = 'field_media_embeddable_oembed';
+    }
+    //$source_field_name = $this->getSourceFieldName($media_type);
+    $media = array_map(function ($source_field_value) use ($media_type, $media_storage, $source_field_name) {
```

Comment:
This class extends the `OEmbedForm` in core, and this stuff is here because this is an override of the `processInputValues()` function in that core class.  This `array_map` call is identical to what's in core.

Reply:
Can you not call the parent method instead of duplicating it here?

Reply:
The only reason I can't call the parent function here is because it will always pick the oembed field as the source field.  I needed to modify this function so that it would use the correct field as the source field in both the case of oembeds and unstructured embeds.  I'd have to pass that value in to the parent method, which would mean modifying core.